src/thunderbots/software/grsim_communication/grsim_command_primitive_visitor.cpp

```diff
@@ -148,7 +148,69 @@ void GrsimCommandPrimitiveVisitor::visit(const MoveSpinPrimitive &move_spin_prim
 
 void GrsimCommandPrimitiveVisitor::visit(const PivotPrimitive &pivot_primitive)
 {
-    // TODO: https://github.com/UBC-Thunderbots/Software/issues/94
+    // Compute final position
+    Point final_robot_position(
+        pivot_primitive.getPivotPoint().x() +
+            (pivot_primitive.getPivotRadius() * pivot_primitive.getFinalAngle().cos()),
+        pivot_primitive.getPivotPoint().y() +
+            (pivot_primitive.getPivotRadius() * pivot_primitive.getFinalAngle().sin()));
+
+    // Get unit vector from pivot point to robot
+    Vector unit_pivot_point_to_robot =
+        (pivot_primitive.getPivotPoint() - robot.position()).norm();
+
+    // find the general direction to travel, project those onto the two possible
+    // directions and see which one is better
+    Vector general_direction_to_destination = final_robot_position - robot.position();
+
+    // in the primitive, both tangential directions are dotted with a vector from the
+    // robot to the final position to get the shortest way to rotate (CW/CCW). This
+    // direction is computed once and does not change throughout the lifetime of the
+    // primitive. There is no equivalent setup function in the
+    // GrsimCommandPrimitiveVisitor class, so a hardcoded direction is chosen.
```

Comment:
I may be wrong here, but I think that if you choose the best direction and start moving in that direction, then at any later point the best direction will always be the direction you are already moving in? So you wouldn't need to hard-code this here?

Reply:
I had an issue where, when it was picking the direction to move, it would reach a stalemate when there was wobble trying to get back onto orbit and then it decided that going the other way was better and would start rotating the other way. It would then wobble slightly out of orbit again and decide to change its mind and go the other way. It would just get into this weird bobble. If you pick the best direction at the start and stick to it, it works really well, or if you can get multiple accelerations and sum them and apply them, it will get rid of that janky step where it stops rotating to get back onto orbit. It would instead add the correction to its rotation and be incredibly smooth, which would allow it to not go into a wobble and be indecisive about the direction it chooses.

Reply:
If I'm reading this correctly, looks like this is no longer hard coded? Is the wobbling no longer an issue?

Reply:
Yeah thats correct, I changed the way on how it resolves which direction to move, its different from how the primitive does it, but it works better in grsim. The primitive way didn't work here.